app/src/main/java/com/github/siela1915/bootcamp/FragmentIngredientCheck.java

```diff
@@ -0,0 +1,124 @@
+    List<Ingredient> ingredientList= Collections.emptyList();
+    private RecyclerView recyclerView;
+    private Button shoppingCartBtn,nearbyBtn;
+    private IngredientCheckAdapter adapter;
+    private ShoppingListManager shoppingManager;
+
+    public FragmentIngredientCheck() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     //* @param param1 Parameter 1.
+     //* @param param2 Parameter 2.
+     * @return A new instance of fragment FragmentIngredientCheck.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static FragmentIngredientCheck newInstance(List<Ingredient> ingredients) {
+        FragmentIngredientCheck fragment = new FragmentIngredientCheck();
+        Bundle args = new Bundle();
+        //args.putString(ARG_PARAM1, param1);
+        //args.putString(ARG_PARAM2, param2);
+        args.putParcelableArrayList(ARG_INGREDIENT_LIST, new ArrayList<>(ingredients));
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            ingredientList = getArguments().getParcelableArrayList(ARG_INGREDIENT_LIST);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_ingredient_check, container, false);
+        recyclerView = view.findViewById(R.id.neededIngredientsRV);
+        recyclerView.setHasFixedSize(true);
+        recyclerView.setLayoutManager(new GridLayoutManager(getContext(),2));
+        List<String> ingList = extractNameOfIngredient(ingredientList);
+        adapter = new IngredientCheckAdapter(ingList);
+        recyclerView.setAdapter(adapter);
+        shoppingCartBtn = view.findViewById(R.id.addToShoppingListBtn);
+        nearbyBtn=view.findViewById(R.id.nearByBtn);
+        shoppingManager = new ShoppingListManager(getContext());
+        shoppingCartBtn.setOnClickListener(v -> {
+            for(String item: adapter.getSelectedItems()){
+                shoppingManager.addIngredient(item);
+            }
+            ShoppingCartFragment fragment = new ShoppingCartFragment();
+            getParentFragmentManager().beginTransaction()
+                    .replace(R.id.ingredientContainer,fragment)
+                    .addToBackStack(null)
+                    .commit();
+        });
+        nearbyBtn.setOnClickListener(v->{
+            NearbyHelpFragment fragment= new NearbyHelpFragment();
+            getParentFragmentManager().beginTransaction()
+                    .replace(R.id.ingredientContainer,fragment)
+                    .addToBackStack(null)
+                    .commit();
+        });
```

Comment:
This logic seems to repeat itself, might consider creating a separate method as `onCreateView` method is also not very short.